app/src/entities/areaV2.entity.js

```diff
@@ -0,0 +1,121 @@
+    isAdministrativeBoundary() {
+        const hasIso = Boolean(this.areaModel.iso?.country);
+        const hasAdmin = Boolean(this.areaModel.admin?.adm0);
+        return hasIso || hasAdmin;
+    }
+
+    /**
+     * Extract and return an array of administrative names from the `areaModel.name` property.
+     * The names are split by commas, reversed, and trimmed. This is so the order of the names matches the
+     * order of IDs.
+     * For example, "State, Country" becomes ["Country", "State"].
+     *
+     * @returns {string[]} Array of reversed, trimmed administrative names (empty if no name is provided).
+     */
+    gatherAdministrativeNames() {
+        // Safely create an array of trimmed names, reversed.
+        return this.areaModel.name
+            ? this.areaModel.name.split(',').reverse().map((v) => v.trim())
+            : [];
+    }
+
+    /**
+     * Gather the country, region, and subregion IDs based on the `admin` or `iso` properties.
+     * - If `admin.adm0` is present, use `adm0` (country), `adm1` (region), and `adm2` (subregion).
+     *   - Region and subregion may be numeric, so we pass them through `safeNumberToString`.
+     * - Otherwise, use `iso.country`, `iso.region`, and `iso.subregion`.
+     *
+     * @returns {[string|null, string|null, string|null]} A tuple of `[countryId, regionId, subregionId]`.
+     */
+    gatherAdministrativeIds() {
+        let countryId;
+        let regionId;
+        let subregionId;
+
+        if (this.areaModel.admin?.adm0) {
+            const { admin: { adm0, adm1, adm2 } } = this.areaModel;
+            countryId = adm0?.trim() ?? null;
+            regionId = AreaEntity.safeNumberToString(adm1);
+            subregionId = AreaEntity.safeNumberToString(adm2);
+        } else {
+            const { iso: { country, region, subregion } } = this.areaModel;
+            countryId = country?.trim() ?? null;
+            regionId = region?.trim() ?? null;
+            subregionId = subregion?.trim() ?? null;
+        }
+
+        return [countryId, regionId, subregionId];
+    }
+
+    /**
+     * Build and add (or update) an AdministrativeVersion entry in `areaModel.adminVersions`.
+     * Will only proceed if the area is an administrative boundary.
+     */
+    populateAdminVersions() {
+        if (!this.isAdministrativeBoundary()) return;
+
+        const adminVersion = AdministrativeVersion.build(
+            this.areaModel.geostore,
+            this.gatherAdministrativeNames(),
+            this.gatherAdministrativeIds(),
+        );
+
+        this.addAdminVersion(adminVersion);
+    }
+
+    /**
+     * Insert or replace an `AdministrativeVersion` in the `adminVersions` array.
+     * - If the version does not already exist, it is appended.
+     * - If it does exist, the existing entry is replaced in place.
+     *
+     * @param {AdministrativeVersion} adminVersion - The new or updated AdministrativeVersion instance.
+     */
+    addAdminVersion(adminVersion) {
```

Comment:
Not finding adminVersions being instantiated with empty array... will mongoose know to create one when there is no existing record to append to here?

Reply:
Thank you for reading through the tests, @solomon-negusse!
I was thinking, "_I'm pretty sure I wrote a test for that..._" when I saw the first version of your comment. 😄